ros/resources/v0/hosts.py

```diff
@@ -0,0 +1,67 @@
+import json
+from flask import jsonify
+from flask import request, Response
+from flask_restful import Resource, abort, fields, marshal_with
+from marshmallow import Schema, fields
+
+from ros.models import PerformanceProfile
+from ros.utils import is_valid_uuid
+from ros.lib.host_inventory_interface import fetch_all_hosts_from_inventory
+
+import logging
+
+class PerformanceProfileSchema(Schema):
+    inventory_id = fields.Str()
+    performance_record = fields.Dict(keys=fields.Str(), values=fields.Str())
+    performance_score = fields.Dict(keys=fields.Str(), values=fields.Int())
+
+class HostsApi(Resource):
+    def get(self):
```

Comment:
```suggestion
class HostsApi(Resource):
    facts_fields = {
        'cloud_provider': fields.String,
        'instance_type': fields.String,
        'idling_time': fields.String,
        'io_wait': fields.String
    }
    performance_score_fields = {
        'cpu_score': fields.Integer,
        'memory_score': fields.Integer,
        'io_score': fields.Integer
    }
    hosts_fields = {
        'fqdn': fields.String,
        'display_name': fields.String,
        'id': fields.String,
        'account': fields.String,
        'recommendation_count': fields.Integer,
        'state': fields.String,
        'performance_score': fields.Nested(performance_score_fields),
        'facts': fields.Nested(facts_fields)

    }
    output_fields = {
        'results': fields.List(fields.Nested(hosts_fields))
    }

    @marshal_with(output_fields)
    def get(self):
```

Reply:
Doesn't it expect to have a model for the respective resource class we are defining these fields? For eg. in case of `fact_fields`, would it understand the 4 fields and consider them adding to the response as we do not have any model defined for the same or am I missing something here?

Reply:
> Doesn't it expect to have a model for the respective resource class we are defining these fields?

No, marshalling the object has no relation with the model. Consider the above fields as validation of our response. Explanation - the return object `hosts` of this function contains many unnecessary(sometimes sensitive info). Using `@marshal_with` only includes the fields in response which we have specified.